tests/pybind11_tests.cpp

```diff
@@ -80,10 +80,10 @@ PYBIND11_MODULE(pybind11_tests, m) {
 
     // Intentionally kept minimal to not create a maintenance chore
     // ("just enough" to be conclusive).
-#if defined(_MSC_FULL_VER)
-    m.attr("compiler_info") = "MSVC " PYBIND11_TOSTRING(_MSC_FULL_VER);
-#elif defined(__VERSION__)
+#if defined(__VERSION__)
```

Comment:
I think it's unfortunate either way around.

I didn't realize before that Windows Clang has this problem. It's also not great as is.

Ideally I think we'd have something like this:

```
#if defined(_MSC_FULL_VER)
#    if defined(__VERSION__)
        m.attr("compiler_info") = __VERSION__ " MSVC " PYBIND11_TOSTRING(_MSC_FULL_VER);
#    else
        m.attr("compiler_info") = "MSVC " PYBIND11_TOSTRING(_MSC_FULL_VER);
#    endif
#elif defined(__VERSION__)
    m.attr("compiler_info") = __VERSION__;
```

Reply:
`__version__` is a string that describes the compiler (name and version), and it's not provided by MSVC. So if `__version__` is set, use that, otherwise provide the fallback. I don't see a reason to add a case that's not possible (from what I understand).

Reply:
I checked a variety of MSVC builds, they all print correctly.